13_07_2017/3D_slab_modes.py

```diff
@@ -0,0 +1,779 @@
+    if 'slow' in mode and 'body-2' in mode:
+        slow_body_2_mode_options.append(mode)
+    if 'slow' in mode and 'body-3' in mode:
+        slow_body_3_mode_options.append(mode)
+    if 'slow' in mode and 'body' in mode:
+        slow_body_mode_options.append(mode)
+
+# Which angle shall we view from? OPTIONS:
+view_options = ['front', 'front-parallel', 'top', 'top-parallel', 'front-top',
+                'front-side', 'front-top-side']
+
+# Uniform lighting?
+#uniform_light = True
+uniform_light = False
+
+show_density = False
+show_density_pert = False
+show_mag = False
+show_mag_scale = False
+show_mag_fade = False
+show_mag_vec = False
+show_vel_front = False
+show_vel_front_pert = False
+show_vel_top = False
+show_vel_top_pert = False
+show_disp_top = False
+show_disp_front = False
+show_axes = False
+show_axis_labels = False
+show_mini_axis = False
+show_boundary = False
+
+# Uncomment the parametrer you would like to see
+# No density perturbations or vel/disp pert for alfven modes.
+
+#show_density = True
+#show_density_pert = True
+show_mag = True
+#show_mag_scale = True #must also have show_mag = True
+#show_mag_fade = True
+#show_mag_vec = True
+#show_vel_front = True
+#show_vel_front_pert = True
+#show_vel_top = True
+show_vel_top_pert = True
+#show_disp_top = True
+#show_disp_front = True
+show_axes = True
+#show_axis_labels = True
+show_mini_axis = True
+show_boundary = True
+
+# Visualisation modules in string form for file-names
+vis_modules = [show_density, show_density_pert, show_mag, show_mag_scale, 
+               show_mag_fade, show_mag_vec, show_vel_front, show_vel_front_pert,
+               show_vel_top, show_vel_top_pert, show_disp_top, show_disp_front]
+vis_modules_strings = ['show_density', 'show_density_pert', 'show_mag', 'show_mag_scale', 
+                       'show_mag_fade', 'show_mag_vec', 'show_vel_front', 'show_vel_front_pert',
+                       'show_vel_top', 'show_vel_top_pert', 'show_disp_top', 'show_disp_front']
+vis_mod_string = ''
+for i in range(len(vis_modules)):
+    if vis_modules[i] == True:
+        vis_mod_string = vis_mod_string + vis_modules_strings[i][5:] + '_'
+
+
+# Set to True if you would like the dispersion diagram with chosen mode highlighted.
+show_dispersion = False
+#show_dispersion = True
+
+# Wanna see the animation? Of course you do
+#show_animation = False
+show_animation = True
+
+# Basic plot to see which eigensolutions have been found.
+show_quick_plot = False
+#show_quick_plot = True
+
+# Video resolution
+#res = (1920,1080) # There is a problem with this resolution- height must be odd number - Mayavi bug apparently
+res = tuple(101 * np.array((16,9)))
+#res = tuple(51 * np.array((16,9)))
+#res = tuple(21 * np.array((16,9)))
+
+number_of_frames = 50
+
+# Frames per second of output video
+fps = 20
+
+#save_images = False
+save_images = True
+
+#make_video = False
+make_video = True
+
+#
+##
+###
+####
+#####
+######
+#######
+########
+#########
```

Comment:
Separates the 'preamble' from the body of the code. Is there a more common/standard way to do this? Or is it not something that is recommended?